### PR TITLE
Test role adds and deletes test data

### PR DESCRIPTION
**Pull request by alf-cactus**
Test users are added with a random string. These are deleted after testing or upon a failure
PW length increased

### CODE REVIEW

**Review by tpurschke — CHANGES_REQUESTED** [review]
- version number is still 5.4.2 - you need to merge current develop branch
- fresh install works
- upgrade fails with error
```
TASK [test : middleware test get jwt valid creds] ***************************************************************************************************************************
fatal: [fworch-srv]: FAILED! => changed=false 
  connection: close
  content: '{"error":"A0002 Invalid credentials"}'
  content_length: '37'
  date: Mon, 29 Nov 2021 09:22:10 GMT
  elapsed: 0
  msg: 'Status code was 400 and not [200]: HTTP Error 400: Bad Request'
  redirected: false
  server: Microsoft-NetCore/2.0
  status: 400
  url: https://127.0.0.1:8888/AuthenticateUser/
```

**Comment by tpurschke** [comment]
after install & upgrade, the following test data was left behind (only in LDAP):
```console
fworch@debian11:~/etc/secrets$ ldapsearch -x -W -D cn=manager,ou=systemuser,ou=user,dc=fworch,dc=internal -y /usr/local/fworch/etc/secrets/ldap_manager_pw.txt "(ou=*_test_*)"
# extended LDIF
#
# LDAPv3
# base <dc=fworch,dc=internal> (default) with scope subtree
# filter: (ou=*_test_*)
# requesting: ALL
#

# tenant1_test_mGrZSKMlZZWCfoByHYxnswgvevdMtjtf, operator, user, fworch.interna
 l
dn: ou=tenant1_test_mGrZSKMlZZWCfoByHYxnswgvevdMtjtf,ou=operator,ou=user,dc=fw
 orch,dc=internal
objectClass: top
objectClass: organizationalUnit
ou: tenant1_test_mGrZSKMlZZWCfoByHYxnswgvevdMtjtf

# tenant2_test_mGrZSKMlZZWCfoByHYxnswgvevdMtjtf, operator, user, fworch.interna
 l
dn: ou=tenant2_test_mGrZSKMlZZWCfoByHYxnswgvevdMtjtf,ou=operator,ou=user,dc=fw
 orch,dc=internal
objectClass: top
objectClass: organizationalUnit
ou: tenant2_test_mGrZSKMlZZWCfoByHYxnswgvevdMtjtf
```

File: roles/sample-auth-data/tasks/auth_sample_data.yml [hunk 1]
```diff
@@ -1,49 +1,50 @@
...
         DO $do$ BEGIN 
           IF NOT EXISTS 
-            (SELECT * FROM tenant_to_device LEFT JOIN tenant USING (tenant_id) WHERE tenant_name='tenant2_demo') THEN 
+            (SELECT * FROM tenant_to_device LEFT JOIN tenant USING (tenant_id) WHERE tenant_name='tenant2_{{ sample_postfix }}') THEN 
             INSERT INTO tenant_to_device (tenant_id, device_id) 
-          SELECT tenant_id, (select dev_id FROM device where device.dev_name='checkpoint_demo') from tenant WHERE tenant.tenant_name='tenant2_demo'; 
+          SELECT tenant_id, (select dev_id FROM device where device.dev_name='checkpoint_{{ sample_postfix }}') from tenant WHERE tenant.tenant_name='tenant2_{{ sample_postfix }}'; 
           END IF; 
         END $do$
+    when: sample_role_purpose is not match('test')
```
[the remark this answers is not in the document]
Comment (alf-cactus) [reply roles/sample-auth-data/tasks/auth_sample_data.yml:47]: The role should work normally, even if someone changes sample_role_purpose

File: roles/test/tasks/test-auth.yml [hunk 2]
```diff
@@ -9,7 +9,7 @@
     timeout: 25
   when: "not run_on_github|bool"
 
-# curl --request POST --url http://127.0.0.1:8888/AuthenticateUser/ --header 'content-type: application/json' --data '{"Username":"user1_demo","Password":"cactus1"}'
+# curl --request POST --url http://127.0.0.1:8888/AuthenticateUser/ --header 'content-type: application/json' --data '{"Username":"user1_{{ test_postfix }}","Password":"cactus1"}'
```
[the remark this answers is not in the document]
Comment (alf-cactus) [reply roles/test/tasks/test-auth.yml]: done

**Comment by alf-cactus** [comment]
had a wrong filter for deleting ldap data, should work now

File: inventory/group_vars/sampleserver.yml [hunk 3]
```diff
@@ -7,6 +7,7 @@ importer_hostname: localhost
 sample_data_rate: medium
 sample_role_purpose: demo
 sample_fortigate_name: fortigate_demo
+sample_postfix: demo
```
[the remark this answers is not in the document]
Comment (alf-cactus) [reply inventory/group_vars/sampleserver.yml:10]: done

File: roles/test/tasks/main.yml [hunk 4]
```diff
@@ -12,6 +12,39 @@
   become: yes
   environment: "{{ proxy_env }}"
 
+- name: randomize test names to avoid interference with production data
+  set_fact:
+    test_postfix: "test_{{ random_generated_pw }}"
```
[the remark this answers is not in the document]
Comment (alf-cactus) [reply roles/test/tasks/main.yml:17]: done

**Review by tpurschke — CHANGES_REQUESTED** [review]
- test fresh install worked ok
- upgrade directly afterwards threw error:
```
TASK [test : middleware test get jwt valid creds] *****************************************************************************************************************
fatal: [fworch-srv]: FAILED! => changed=false 
  connection: close
  content: A0002 Invalid credentials
  content_type: text/plain; charset=utf-8
  date: Fri, 03 Dec 2021 14:45:23 GMT
  elapsed: 0
  msg: 'Status code was 400 and not [200]: HTTP Error 400: Bad Request'
  redirected: false
  server: Kestrel
  status: 400
  transfer_encoding: chunked
  url: https://127.0.0.1:8888/api/AuthenticationToken/Get/

RUNNING HANDLER [test : delete test user and dir] *****************************************************************************************************************
ok: [fworch-srv]
```
even login with admin user was not possible anymore after this (invalid creds)
but direct ldap access works:
```
tim@debian11:~$ sudo -u fworch ldapwhoami -x -D uid=admin,ou=tenant0,ou=operator,ou=user,dc=fworch,dc=internal -w fworch.1
dn:uid=admin,ou=tenant0,ou=operator,ou=user,dc=fworch,dc=internal
tim@debian11:~$ 

```

**Review by tpurschke — APPROVED** [review]
install & upgrade tested successfully
no left-overs from test install found
perfect!

**Pull request merged**